FogCreek/Glitch-Community
# Piquant mountain: Project Options Pop etc.

depends on #566,  #567, #596 

## Links
* Remix link: https://piquant-mountain.glitch.me/
* Manuscript link: https://glitch.manuscript.com/f/cases/3328622/

## GIF/Screenshots:
![Screen Shot 2019-05-24 at 3 36 49 PM](https://user-images.githubusercontent.com/938722/58353655-16139e00-7e3d-11e9-85a4-58613d356478.png)
![Screen Shot 2019-05-24 at 3 49 30 PM](https://user-images.githubusercontent.com/938722/58353656-16139e00-7e3d-11e9-9b86-c8c3364cc00f.png)
![Screen Shot 2019-05-24 at 3 50 09 PM](https://user-images.githubusercontent.com/938722/58353657-16139e00-7e3d-11e9-84a1-febf06e5beb6.png)
![Screen Shot 2019-05-24 at 3 50 21 PM](https://user-images.githubusercontent.com/938722/58353658-16139e00-7e3d-11e9-938e-2186be8f01b4.png)
![Screen Shot 2019-05-24 at 3 50 47 PM](https://user-images.githubusercontent.com/938722/58353659-16139e00-7e3d-11e9-9cfb-f3c80120560d.png)
![Screen Shot 2019-05-24 at 3 50 58 PM](https://user-images.githubusercontent.com/938722/58353660-16139e00-7e3d-11e9-9a49-4c7d4064f9b4.png)
![Screen Shot 2019-05-24 at 3 51 55 PM](https://user-images.githubusercontent.com/938722/58353661-16139e00-7e3d-11e9-98c3-dccae57cb9ec.png)

## Changes:
* componentize ProjectOptionsPop
* componentize AddProjectToCollectionPop
* use Algolia for searching collections
* make collection results always searchable & navigable from keyboard
* componentize CreateCollectionPop
* use CreateCollectionPop for the Create Collection button on profile pages
* add short cache to API requests

## How To Test:
+ log into https://piquant-mountain.glitch.me/
  * visit your profile page
  * clicking on a project's options menu should show the project options popover
  * clicking on "Add to collection" should open a larger popover that lists collections to which you can add this project
  * typing in the search field should show filtered results
  * using the arrow keys should navigate the projects list
  * clicking on a collection result, or hitting enter when one is selected, should add the project to that collection
  * clicking on "Add to a new collection" should show the create collection popover
  * after entering a collection name, clicking "Create Collection" should create a collection with the selected project in it
+ visit your profile page
  * clicking on Create Collection should show the create collection popover (without the team selection dropdown)
  * after entering a collection name, clicking "Create Collection" should create a collection and redirect you to the page
